src/app/shared/form/copy-input.directive.ts

```diff
@@ -0,0 +1,36 @@
+import { Directive, ElementRef, Input, OnInit, HostListener } from '@angular/core';
+
+@Directive({
+  selector: '[publishCopyInput]'
+})
+export class CopyInputDirective implements OnInit {
+
+  @Input() publishCopyInput: HTMLInputElement;
+
+  private startText = 'Copy';
+  private copiedText = 'Copied';
+
+  constructor(private el: ElementRef) {}
+
+  @HostListener('mouseover') onMouseOver(): void {
+    this.el.nativeElement.innerHTML = this.startText;
+  }
+
+  @HostListener('click') onClick(): void {
+    if (this.publishCopyInput && this.publishCopyInput.select) {
```

Comment:
From what I recall, with this same code in play.prx.org TypeScript thought that publishCopyInput was of type HTMLAnchorElement (even though we know it's a HTMLInputElement) and was throwing console errors because HTMLAnchorElement did not have a select() function. Not sure if it is complaining about this cuz I can't run it right now...